code401Challenges/src/main/java/code401Challenges/linkedlist/LinkedList.java

```diff
@@ -109,5 +109,28 @@ public Boolean includes(int value) {
         }
         return false;
     }
+
+    //=== Find a value in the list that is k nodes from the end ===
+    public int runningBackwards(int k) {
+        Node curr = head;
+        int listLength = 0;
+
+        while (curr != null) {
+            listLength += 1;
+            curr = curr.next;
+        }
+        int stepsToTake = listLength - k;
+        if(stepsToTake < 0 || stepsToTake > listLength)
+            throw new IllegalArgumentException("k is too big, given the length of the list");
```

Comment:
I love that you're throwing an IllegalArgumentException here! Double check the message it gives.